### PR TITLE
Hacktoberfest 2020: Add style improvements to solutions for Project Euler Problem 04

### PR DESCRIPTION
**Pull request by archaengel**
### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

Related #2786

### CODE REVIEW

**Comment by dhruvmanila** [comment]
Please don't remove newlines blindly. If they were already present let them be.

**Comment by archaengel** [comment]
Thanks for taking a look! I'll restore them.

File: project_euler/problem_04/sol1.py [hunk 1]
```diff
@@ -8,44 +8,33 @@
...
                 divisor -= 1
-    raise ValueError("That number is larger than our acceptable range.")
+    return -1
 
 
 if __name__ == "__main__":
-    import doctest
-
-    doctest.testmod()
-
```
Comment (archaengel) [inline project_euler/problem_04/sol1.py:50]: Is this something we'd like to leave in the solutions if it's already present?

File: project_euler/problem_04/sol1.py [hunk 2]
```diff
@@ -40,12 +38,8 @@ def solution(n):
                 if (number % divisor == 0) and (len(str(number // divisor)) == 3.0):
                     return number
                 divisor -= 1
-    raise ValueError("That number is larger than our acceptable range.")
+    return -1
```
Comment (dhruvmanila) [inline project_euler/problem_04/sol1.py:41]: Um, why are you changing the code?
The main goal for this PR is just to fix type hints and default arguments. Please don't make it hard for us to review

File: project_euler/problem_04/sol1.py (hunk 1 above)
Comment (dhruvmanila) [reply project_euler/problem_04/sol1.py:50]: Yes, as I mentioned please don't change anything except for the type hints and default argument values.

File: project_euler/problem_04/sol1.py (hunk 2 above)
Comment (archaengel) [reply project_euler/problem_04/sol1.py:41]: My mistake, I'll scope my changes down to the default arguments and type hints.

File: project_euler/problem_04/sol2.py [hunk 3]
```diff
@@ -10,24 +10,22 @@
...
     39893
-    >>> solution(10000)
-    -1
     """
     answer = 0
     for i in range(999, 99, -1):  # 3 digit numbers range from 999 down to 100
         for j in range(999, 99, -1):
-            product_string = str(i * j)
-            if product_string == product_string[::-1] and i * j < n:
+            t = str(i * j)
```
Comment (dhruvmanila) [inline project_euler/problem_04/sol2.py]: Expanding variable names is fine.

Comment (archaengel) [reply project_euler/problem_04/sol2.py]: Gotcha, I'll add the expanded variable names shortly.

**Review by dhruvmanila — APPROVED** [review]
Thank you for your contribution!

**Comment by archaengel** [comment]
Thanks for taking the time to review!

**Pull request merged**